### ISSUE TITLE
Settings abstraction

### ISSUE THREAD

**Issue by FichteFoll** [SublimeText/sublime_lib#2]
An abstractions of the built-in `sublime.Settings` class seems in order, as the following features would be quite useful:

- [x] Registering a callback that is only called when settings of interest *actually* changed, not when the files were reloaded and *could have* changed.
- [x] Providing a default value at instantiation or at a static location (e.g. through `sublime.decode_value(sublime.load_resource(path_to_bundled_settings_file))` rather than having to repeat the default every time the value is accessed.
- [x] Ability to determine *which* setting changed exactly and what the previous value was, if desired.
- [x] Should function independently of whether the underlying settings object is a view or a file settings object.

A related implementation can be found here: https://github.com/SublimeText/InactivePanes/blob/ca417aa6699432ffbbd716ea44e8d431a8d5818f/settings/__init__.py
Downside of that implementation is the complex attribute-to-setting-name mapping that shouldn't need to exist.

**Comment by Thom1729** [comment]
Can't default values be handled by bundling a default `Name.sublime-settings` file with the package?

**Comment by Thom1729** [comment]
As of #15, would you agree that we've handled the known use cases?

**Comment by FichteFoll** [comment]
Agreed. All the points are checked, so this should be finished until we find something else, which would be a new issue.
